oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -157,6 +160,46 @@ public AccessToken refreshAccessToken() throws IOException {
     return new AccessToken(accessToken, new Date(expiresAtMilliseconds));
   }
 
+  /**
+   * Returns a Google ID Token from the metadata server on ComputeEngine
+   *
+   * @param targetAudience the aud: field the IdToken should include
+   * @param options list of Credential specific options for the token. For example, an IDToken
+   *     for a ComputeEngineCredential could have the full formatted claims returned if
+   *     IdTokenProvider.Option.FORMAT_FULL) is provided as a list option. Valid option values are:
+   *     <br>
+   *     IdTokenProvider.Option.FORMAT_FULL<br>
+   *     IdTokenProvider.Option.LICENSES_TRUE<br>
+   *     If no options are set, the defaults are "&amp;format=standard&amp;licenses=false"
+   * @throws IOException if the attempt to get an IdToken failed
+   * @return IdToken object which includes the raw id_token, JsonWebSignature
+   */
+  @Override
+  public IdToken idTokenWithAudience(String targetAudience, List<IdTokenProvider.Option> options)
+      throws IOException {
+    GenericUrl documentUrl = new GenericUrl(getIdentityDocumentUrl());
+    if (options != null) {
+      if (options.contains(IdTokenProvider.Option.FORMAT_FULL)) {
+        documentUrl.set("format", "full");
+      }
+      if (options.contains(IdTokenProvider.Option.LICENSES_TRUE)) {
+        // license will only get returned if format is also full
+        documentUrl.set("format", "full");
+        documentUrl.set("license", "TRUE");
+      }
+    }
+    documentUrl.set("audience", targetAudience);
+    HttpResponse response = null;
+    response = getMetadataResponse(documentUrl.toString());
+    InputStream content = response.getContent();
+    if (content == null) {
+      throw new IOException("Empty content from metadata token server request.");
+    }
+    String rawToken = response.parseAsString();
+    JsonWebSignature jws = JsonWebSignature.parse(OAuth2Utils.JSON_FACTORY, rawToken);
+    return new IdToken(rawToken, jws);
```

Comment:
No longer need to do the parsing here.

```suggestion
    return IdToken.create(rawToken);
```